Doc/add ability system workarounds (#621)
spatialos/UnrealGDK: SpatialGDK/Documentation/content/ability-system.md

# Gameplay Ability System on the GDK for Unreal

> The [Gameplay Ability System](https://docs.unrealengine.com/en-us/Gameplay/GameplayAbilitySystem) is a highly flexible framework for building abilities and attributes of the type you might find in an RPG, MMO, or MOBA. However, as it makes use of lots of the advanced parts of the Unreal networking stack, the GDK for Unreal doesn't *currently* support it in it's entirity. On this page we'll list the workarounds that are required if you want to use the Ability System until we have more complete parity with native Unreal.

Reviewer: ```suggestion
The [Gameplay Ability System](https://docs.unrealengine.com/en-us/Gameplay/GameplayAbilitySystem) is a highly flexible framework for building abilities and attributes of the type you might find in an RPG, MMO, or MOBA. However, as it makes use of lots of the advanced parts of the Unreal networking stack, the GDK for Unreal doesn't *currently* support it in it's entirety. On this page we'll list the workarounds that are required if you want to use the Ability System with the Alpha release of the GDK. We are working to remove some of these limitations in the future.
```

## Current Gameplay Ability Workarounds

> 1. Support for Replicated Gameplay Abilities is not supported. Please ensure that any `UGameplayAbility` has its `ReplicationPolicy` set to `ReplicateNo`. Replicated Gameplay Abilities will be supported with dynamic component support

Reviewer: ```suggestion
1. We currently do not support Replicated Gameplay Abilities. Please ensure that any `UGameplayAbility` has its `ReplicationPolicy` set to `ReplicateNo`. Replicated Gameplay Abilities will be supported with dynamic component support
```

Reviewer 2: > Can re remove the "please" from this sentence, please?

> 1. We currently do not support Replicated Gameplay Abilities. Please ensure that any `UGameplayAbility` has its `ReplicationPolicy` set to `ReplicateNo`. Replicated Gameplay Abilities will be available once we support dynamic actor components in the GDK.

Author: ```suggestion
1. We currently do not support Replicated Gameplay Abilities. Ensure that any `UGameplayAbility` has its `ReplicationPolicy` set to `ReplicateNo`. Replicated Gameplay Abilities will be available once we support dynamic actor components in the GDK.
```
